task4-adapter/promisify/solution.js

```diff
@@ -0,0 +1,48 @@
+'use strict';
+const timers = require('node:timers/promises');
+
+// Task: implement a cancelable promise by passing `timeout: number`
+// as an option to the promisified function (last argument,
+// replacing the callback).
+
+const promisify = (fn, options = { timeout: 1000 }) => (...args) => {
```

Comment:
Use default timeout 0 instead of 1000
```suggestion
const promisify = (fn, options = { timeout: 0 }) => (...args) => {
```